shimmingtoolbox/shim/sequencer.py

```diff
@@ -93,12 +93,35 @@ def shim_sequencer(nii_fieldmap, nii_anat, nii_mask_anat, slices, coils: ListCoi
 
     # Make sure anat has the appropriate dimensions
     anat = nii_anat.get_fdata()
-    if anat.ndim != 3:
-        raise ValueError("Anatomical image must be in 3d")
+    if anat.ndim == 3:
+        pass
+    elif anat.ndim == 4:
+        logger.debug("Target anatomical is 4d, taking the average and converting to 3d")
+        anat = np.mean(anat, axis=3)
+        nii_anat = nib.Nifti1Image(anat, nii_anat.affine, header=nii_anat.header)
+    else:
+        raise ValueError("Target anatomical image must be in 3d or 4d")
 
     # Make sure the mask has the appropriate dimensions
-    if nii_mask_anat.get_fdata().ndim != 3:
-        raise ValueError("Mask image must be in 3d")
+    mask = nii_mask_anat.get_fdata()
+    if mask.ndim == 3:
+        pass
+    elif mask.ndim == 4:
```

Comment:
that seems strange to input a 4d mask (why would ppl do that?), and if they do, i'm not sure if averaging across volumes is the solution (eg: why not taking the first mask of the series?).

what were the scenario you had in mind for inputting 4d masks?

Reply:
> that seems strange to input a 4d mask (why would ppl do that?)

If the user thresholds an EPI to create a mask, the mask will be 4d. We should probably not recommend it but I was think the user should probably have the flexibility to do so. 

> i'm not sure if averaging across volumes is the solution (eg: why not taking the first mask of the series?)

I'm averaging then thresholding to go back to a binary mask (for now). The idea was to do a logical_and() of all volumes but I found it included dead voxels, adding everything then thresholding allowed to remove those dead voxels.

Reply:
> If the user thresholds an EPI to create a mask, the mask will be 4d. We should probably not recommend it but I was think the user should probably have the flexibility to do so.

meh... that's not very common practice. I don't mind keeping the option (as you already implemented it), but i was just curious